jax_galsim/bessel.py

```diff
@@ -101,3 +102,15 @@ def si(x):
         - _g_pade(x, x2) * jnp.sin(x),
         _si_small_pade(x, x2),
     )
+
+
+@jax.jit
+def kv(nu, x):
+    """Modified Bessel 2nd kind"""
+    return tfp.substrates.jax.math.bessel_kve(nu * 1.0, x) / jnp.exp(jnp.abs(x))
+
+
+@jax.jit
+def gamma(x):
+    """Gamma(x)"""
+    return jnp.exp(jax.lax.lgamma(x * 1.0))
```

Comment:
Why do you have `* 1.0` in these functions?

Reply:
`test_spergel.py` : 

1. _E       TypeError: lgamma does not accept dtype int64. Accepted dtypes are subtypes of floating._
2. using ` galsim.Spergel(nu=0, half_light_radius=3.5, flux=1.e4)` makes a crash in kv

`test_bessel.py` idem for kv: 
_E         TypeError: Found incompatible dtypes, <class 'numpy.float64'> and <class 'numpy.int64'>. Seen so far: [<class 'numpy.float64'>, <class 'numpy.int64'>]_